kumascript/macros/JSRef.ejs

```diff
@@ -134,110 +137,197 @@ function isPrototypeMemberPage(page) {
-    const resultSubPath      = item.subPath ?? "";
-    const resultConstructors = item.constructors ?? [];
-    const resultProperties   = item.properties ?? [];
-    const resultMethods      = item.methods ?? [];
-    const resultOpen         = item.defaultOpened ?? false;
+  for (const [index, item] of result.entries()) {
+    const {
+      title,
+      subPath,
+      constructors,
+      staticMethods,
+      staticProperties,
+      instanceMethods,
+      instanceProperties,
+      defaultOpened,
+    } = item;
 
     if (index === 1) {
       output += `<li><strong>${text.Inheritance}</strong></li>`;
     }
 
-    const link = web.smartLink(`${slugStdlib}/${resultSubPath}`, null, `<code>${resultTitle}</code>`, null, slugStdlib, "JSRef");
+    const link = web.smartLink(
+      `${slugStdlib}/${subPath}`,
+      null,
+      `<code>${title}</code>`,
+      null,
+      slugStdlib,
+      "JSRef",
+    );
     output += `<li><strong>${link}</strong></li>`;
 
-    if (resultConstructors.length > 0) {
-      output += await buildSublist(resultConstructors, text.Constructor, resultOpen);
+    if (constructors.length > 0) {
+      output += await buildSublist(
+        constructors,
+        text.Constructor,
+        defaultOpened,
+      );
+    }
+    if (staticMethods.length > 0) {
+      output += await buildSublist(
+        staticMethods,
+        text.StaticMethods,
+        defaultOpened,
+      );
+    }
+    if (staticProperties.length > 0) {
+      output += await buildSublist(
+        staticProperties,
+        text.StaticProperties,
+        defaultOpened,
+      );
     }
-    if (resultProperties.length > 0) {
-      output += await buildSublist(resultProperties, text.Properties, resultOpen);
+    if (instanceMethods.length > 0) {
+      output += await buildSublist(
+        instanceMethods,
+        text.InstanceMethods,
+        defaultOpened,
+      );
     }
-    if (resultMethods.length > 0) {
-      output += await buildSublist(resultMethods, text.Methods, resultOpen);
+    if (instanceProperties.length > 0) {
+      output += await buildSublist(
+        instanceProperties,
+        text.InstanceProperties,
+        defaultOpened,
+      );
     }
+  }
 
-    if (index === 0 && group.length > 0) {
-      output += `<li><strong>${text.Related}</strong></li>`;
-      for (const groupItem of group) {
-        const link = web.smartLink(`${slugStdlib}/${groupItem.replace(".", "/")}`, null, `<code>${groupItem}</code>`, null, slugStdlib, "JSRef");
-        output += `<li><strong>${link}</strong></li>`;
-      }
+  if (group.length > 0) {
+    output += `<li><strong>${text.Related}</strong></li>`;
+    for (const groupItem of group) {
+      const link = web.smartLink(
+        `${slugStdlib}/${groupItem.replace(".", "/")}`,
+        null,
+        `<code>${groupItem}</code>`,
+        null,
+        slugStdlib,
+        "JSRef",
+      );
+      output += `<li><strong>${link}</strong></li>`;
```

Comment:
Because the inheritance chain is now proper, there's less need for the "related pages" part (except for `Intl`, where navigation remains a bit awkward). I've thus put it below "inheritance" to avoid obstructing the more structured information.